docs/main.md

```diff
@@ -0,0 +1,224 @@
+
+```php
+$vary_headers = [
+	'P13n-Geo-Country-Code' => true,
+	'P13n-Geo-Country-Name' => false,
+	'P13n-Geo-Region' => false,
+	'P13n-Geo-City' => false,
+	'P13n-Geo-Continent-Code' => false,
+	'P13n-Geo-Conn-Type' => false,
+	'P13n-Geo-Conn-Speed' => false,
+	'P13n-Interest' => true,
+];
+```
+
+#### Example
+To set the country _name_ as the only supported personalization type:
+
+```php
+add_filter( 'pantheon.ei.supported_vary_headers', 'filter_vary_headers' );
+function filter_vary_headers( array $vary_headers ) {
+	$vary_headers['P13n-Geo-Country-Code'] = false;
+	$vary_headers['P13n-Geo-Country-name'] = true;
+	$vary_headers['P13n-Interest'] = false;
+
+	return $vary_headers;
+}
+```
+
+To use the country-based Geolocation data as the only supported personalization type:
+
+```php
+add_filter( 'pantheon.ei.supported_vary_headers', 'filter_vary_headers' );
+function filter_vary_headers( array $vary_headers ) {
+	$vary_headers['P13n-Interest'] = false;
+
+	return $vary_headers;
+}
+```
+
+To set the vary headers to not be sent until consent has been granted (see also [Pantheon Edge Integrations Consent Management](https://github.com/pantheon-systems/pantheon-edge-integrations-consent-management)):
+
+```php
+function check_consent() {
+	if ( ! function_exists( 'wp_has_consent' ) ) {
+		return;
+	}
+
+	if ( ! wp_has_consent() ) {
+		add_filter( 'pantheon.ei.supported_vary_headers', 'do_not_send_vary_headers' );
+	}
+}
+
+function do_not_send_vary_headers() : array {
+	return [
+		'P13n-Geo-Country-Code' => false,
+		'P13n-Interest' => false,
+	];
+}
+```
+
+### `pantheon.ei.custom_header_data`
+
+This filter is applied in the `update_vary_headers` function and allows engineers to modify the custom `HeaderData` before the it's returned.
+
+#### Parameters
+
+_(array)_ `HeaderData` data as an array.
+
+#### Example
+
+```php
+add_filter( 'pantheon.ei.custom_header_data', 'pantheon_ei_customize_header_data' );
+function pantheon_ei_customize_header_data( array $data ) : array {
+	// Remove last element from $data.
+	return array_pop( $data );
+}
+```
+
+### `pantheon.ei.enabled`
+
+Allows developers to filter the output of the `WP\edge_integrations_enabled()` function.
+
+This can be used to force the application to think that the headers have been detected when they haven't.
+
+**Note:** This does not change whether the headers exist, output may be unexpected if this value is "true" but the headers are not present.
+
+#### Parameters
+
+__(bool)__ `$headers_enabled` Whether Edge Integrations have been configured and the CDN is returning data.
+
+#### Example
+
+```php
+// Force Edge Integrations to appear enabled. This will disable the notice that appears when the headers are not detected.
+add_filter( 'pantheon.ei.enabled', '__return_true' );
+
+// Force Edge Integrations to appear disabled. This display the notice that appears when the headers are not detected. It will not change whether Edge Integrations functions will work.
```

Comment:
Adding an s.
```suggestion
// Force Edge Integrations to appear disabled. This displays the notice that appears when the headers are not detected. It will not change whether Edge Integrations functions will work.
```